test/runtests.jl

```diff
@@ -280,15 +280,26 @@ end
   M2 = spzeros(20, 20)
   M2[1:10, 1:10] = A
   M2[11:20, 11:20] = spdiagm(0 => -100ϵ * ones(10))
+  M3 = spzeros(20, 20)
+  M3[1:10, 1:10] = A
+  M3[11:20, 11:20] = spdiagm(0 => zeros(10))
 
   S = ldl_analyze(M1)
   @test S.__analyzed
   @test !S.__factorized
   _allocs1 = @allocated ldl_factorize!(M1, S)
   @test S.d[11:20] ≈ -ϵ * ones(10)
+  @test S.__factorized
   _allocs2 = @allocated ldl_factorize!(M2, S)
   @test S.d[11:20] ≈ -100ϵ * ones(10)
   @test _allocs1 == _allocs2
+
+  @test_throws LDLFactorizations.SQDException ldl_factorize!(M3, S)
+  @test !S.__factorized
```

Comment:
It's fine to access this field internally, but we shouldn't encourage access to it from the outside. That's why I prepended its name with `__`. Maybe there could be a `factorized(S) = S.__factorized` method?

Reply:
Great idea. I added such a function.